nvme_gw_cli.py

```diff
@@ -211,20 +216,74 @@ def create_namespace(self, args):
+        argument("-i", "--nsid", help="Namespace ID", type=int, required=True),
+    ])
+    def delete_namespace(self, args):
+        """Deletes a namespace from a previously created subsystem."""
+
+        try:
+            delete_req = pb2.ns_delete_req(subsystem_nqn=args.subnqn,
+                                           nsid=args.nsid)
+            ret = self.stub.nvmf_subsystem_remove_ns(delete_req)
+            self.logger.info(f"Deleted namespace {delete_req.nsid}: {ret}")
+        except Exception as error:
+            self.logger.error(f"Failed to remove namespace: \n {error}")
+
+    @cli.cmd([
+        argument("-n", "--subnqn", help="Subsystem NQN", required=True),
+        argument("-t", "--host", help="Host NQN", required=True),
+    ])
+    def add_host(self, args):
+        """Adds a host to a previously created subsystem."""
+
+        if args.host == "*":
+            allow_any_host = True
+
+        try:
+            create_req = pb2.subsystem_add_host_req(
+                subsystem_nqn=args.subnqn,
+                host_nqn=args.host,
+                allow_any_host=allow_any_host)
+            ret = self.stub.nvmf_subsystem_add_host(create_req)
+            if allow_any_host:
+                self.logger.info(
+                    f"Allow open host access to {args.subnqn}: {ret.status}")
+            else:
+                self.logger.info(
+                    f"Add host {args.host} access to {args.subnqn}: {ret.status}"
+                )
+        except Exception as error:
+            self.logger.error(f"Failed to add host: \n {error}")
+
+    @cli.cmd([
+        argument("-n", "--subnqn", help="Subsystem NQN", required=True),
+        argument("-t", "--host", help="Host NQN", required=True),
+    ])
+    def delete_host(self, args):
+        """Deletes a host from a previously created subsystem."""
+
+        if args.host == "*":
+            disable_allow_any_host = True
 
         try:
-            allow_req = pb2.subsystem_allow_any_host_req(
-                subsystem_nqn=args.subnqn, disable=0)
-            ret = self.stub.nvmf_subsystem_allow_any_host(allow_req)
-            self.logger.info(f"All host access to {args.subnqn}: {ret.status}")
+            delete_req = pb2.host_delete_req(
+                subsystem_nqn=args.subnqn,
+                host_nqn=args.host,
+                disable_allow_any_host=disable_allow_any_host)
+            ret = self.stub.nvmf_subsystem_remove_host(delete_req)
+            if disable_allow_any_host:
+                self.logger.info(
+                    f"Disable open host access to {args.subnqn}: {ret.status}")
+            else:
+                self.logger.info(
+                    f"Remove host {args.host} access from {args.subnqn}: {ret.status}"
+                )
         except Exception as error:
-            self.logger.error(f"Failed to allow any host: \n {error}")
+            self.logger.error(f"Failed to add host: \n {error}")
```

Comment:
The host is removed here, not added:
```suggestion
            self.logger.error(f"Failed to remove host: \n {error}")
```